Cabal/Distribution/Simple/Program/Ar.hs

```diff
@@ -97,9 +119,26 @@ createArLibArchive verbosity lbi targetPath files = do
   where
     progDb = withPrograms lbi
     Platform hostArch hostOS = hostPlatform lbi
-    verbosityOpts v | v >= deafening = ["-v"]
-                    | v >= verbose   = []
-                    | otherwise      = ["-c"]
+    verbosityOpts v
+      | v >= deafening = ["-v"]
+      | v >= verbose   = []
+      | otherwise      = ["-c"] -- Do not warn if library had to be created.
+
+-- | The @ar@ tool expects response file to contain sequence of strings
+-- delimited by whitespace. Thus, in order to handle file names with spaces
+-- they should be enclosed in single or double quotes.
+--
+-- Windows poses additional challenge to creating correct response files.
+-- Namely, on Windows standard path separator is backspace but in @ar@
+-- response file format it's escape sequence. Therefore backslashes must
+-- be escaped as well.
+escapeFileName :: FilePath -> FilePath
+escapeFileName = concatMap escapeChar
+  where
+    escapeChar :: Char -> String
+    escapeChar '"'  = "\\\""
+    escapeChar '\\' = "\\\\"
+    escapeChar c    = [c]
```

Comment:
That's good catch, however it seems unplausible that file name will contain newlines. I'm not 100% sure that `ar` will like those newlines either.